api/types/resource_test.go

```diff
@@ -0,0 +1,324 @@
+				}, AppSpecV3{
+					PublicAddr: "baz",
+					URI:        "_",
+				})
+				require.NoError(t, err)
+
+				return app
+			},
+		},
+		{
+			name:               "kube cluster",
+			matchingSearchVals: []string{"foo", "prod", "env"},
+			newResource: func() ResourceWithLabels {
+				kc, err := NewKubernetesClusterV3FromLegacyCluster("_", &KubernetesCluster{
+					Name:         "foo",
+					StaticLabels: labels,
+				})
+				require.NoError(t, err)
+
+				return kc
+			},
+		},
+		{
+			name:               "database",
+			matchingSearchVals: []string{"foo", "bar", "baz", "prod", DatabaseTypeRedshift},
+			newResource: func() ResourceWithLabels {
+				db, err := NewDatabaseV3(Metadata{
+					Name:        "foo",
+					Description: "bar",
+					Labels:      labels,
+				}, DatabaseSpecV3{
+					Protocol: "baz",
+					URI:      "_",
+					AWS: AWS{
+						Redshift: Redshift{
+							ClusterID: "_",
+						},
+					},
+				})
+				require.NoError(t, err)
+
+				return db
+			},
+		},
+		{
+			name:               "database with gcp keywords",
+			matchingSearchVals: []string{"cloud", "cloud sql"},
+			newResource: func() ResourceWithLabels {
+				db, err := NewDatabaseV3(Metadata{
+					Name:   "_",
+					Labels: labels,
+				}, DatabaseSpecV3{
+					Protocol: "_",
+					URI:      "_",
+					GCP: GCPCloudSQL{
+						ProjectID: "_",
+					},
+				})
+				require.NoError(t, err)
+
+				return db
+			},
+		},
+		{
+			name:             "app server",
+			searchNotDefined: true,
+			newResource: func() ResourceWithLabels {
+				appServer, err := NewAppServerV3(Metadata{
+					Name: "_",
+				}, AppServerSpecV3{
+					HostID: "_",
+					App:    &AppV3{Metadata: Metadata{Name: "_"}, Spec: AppSpecV3{URI: "_"}},
+				})
+				require.NoError(t, err)
+
+				return appServer
+			},
+		},
+		{
+			name:             "db server",
+			searchNotDefined: true,
+			newResource: func() ResourceWithLabels {
+				dbServer, err := NewDatabaseServerV3(Metadata{
+					Name: "_",
+				}, DatabaseServerSpecV3{
+					HostID:   "_",
+					Hostname: "_",
+				})
+				require.NoError(t, err)
+
+				return dbServer
+			},
+		},
+		{
+			name:             "kube service",
+			searchNotDefined: true,
+			newResource: func() ResourceWithLabels {
+				kubeServer, err := NewServer("_", KindKubeService, ServerSpecV2{})
+				require.NoError(t, err)
+
+				return kubeServer
+			},
+		},
+		{
+			name:             "desktop service",
+			searchNotDefined: true,
+			newResource: func() ResourceWithLabels {
+				desktopService, err := NewWindowsDesktopServiceV3("_", WindowsDesktopServiceSpecV3{
+					Addr:            "_",
+					TeleportVersion: "_",
+				})
+				require.NoError(t, err)
+
+				return desktopService
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			resource := tc.newResource()
+
+			// Nil search values, should always return true
+			match := resource.MatchSearch(nil)
+			require.True(t, match)
```

Comment:
Isn't this covered in [TestMatchSearch](https://github.com/gravitational/teleport/pull/9892/files#diff-ff8e87df6e542d679e3527fd84b678d92e6de72e50c456ab7f28f27cbe48b472R72)?

Reply:
it's slightly different in that `TestMatchSearch` calls the func directly, whereas here we are calling the resources method which then calls the func